_posts/2025-05-05-plan-execute-agent.md

```diff
@@ -0,0 +1,423 @@
+---
+layout: post
+title: "OpenSearch 3.0: Observability meets Agentic AI"
+authors:
+  - pyek-bot
+  - ylwu
+date: 2025-05-05
+categories:
+  - technical-posts
+meta_keywords: Plan and Execute, OpenSearch, AI, Agents, Function calling, Claude
+meta_description: Learn about OpenSearch 3.0's Agent Framework Enhancements and utilize the plan-execute-reflect agent to resolve an observability use case
+---
+OpenSearch 3.0 introduces the [Plan–Execute–Reflect](https://docs.opensearch.org/docs/latest/ml-commons-plugin/agents-tools/agents/plan-execute-reflect/) agent—a powerful new capability that breaks down complex problems, selects and executes tools autonomously, and adapts through reflection. In this post, we’ll show how this agent automates root cause analysis in observability workflows.
+
+## Introduction
+
+In the era of AI, agents have emerged as a powerful way to make Large Language Models (LLMs) more practical and useful. By connecting LLMs with specific tools and capabilities, agents can transform natural language requests into concrete actions and meaningful results. 
+
+OpenSearch ML Commons has been at the forefront of this transformation, providing a robust agent framework that allows users to orchestrate tools and automate workflows. With OpenSearch 3.0, we're taking a significant leap forward by introducing the Plan-Execute-Reflect agent designed for complex, multi-step reasoning tasks.
+
+## What's new in OpenSearch 3.0's agent framework?
+
+* A Plan-Execute-Reflect agent for executing complex multi-step tasks.
+* Native MCP (Model Coordination Protocol) support for seamless integration with external AI tools.
+* Enhanced tool selection and execution capabilities.
+* Asynchronous execution support for long-running tasks.
+
+In this blog, we'll showcase the power of this new agent through a practical observability use case: Automatically investigating a service failure in a microservices application. 
+
+You'll see how the agent can:
+
+* Break down a complex investigation into logical steps
+* Automatically select and use appropriate tools
+* Analyze multiple data sources (logs, traces, metrics)
+* Provide clear, actionable insights
+
+Whether you're managing cloud infrastructure, developing applications, or building AI-powered solutions, this new capability can help streamline your workflows and enhance your problem-solving capabilities. Let's dive in and see how it works.
```

Comment:
"Let's dive in and see how it works."

This sentence seems duplicate with next paragraph.